doc/conf.py

```diff
@@ -776,7 +776,7 @@ def fix_sklearn_inherited_docstrings(app, what, name, obj, options, lines):
 # Theme options are theme-specific and customize the look and feel of a theme
 # further.  For a list of options available for each theme, see the
 # documentation.
-switcher_version_match = "dev" if ".dev" in version else version
+switcher_version_match = "dev" if ".dev" in release else version
```

Comment:
not sure when this changed / stopped working, but above, `version` is defined as X.Y and `release` is the full `__version__` string (incl. the `.devZZZZZZZ` part, if present)